src/pudl/transform/ferc1.py

```diff
@@ -853,6 +964,228 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
         )
 
 
+class PlantInServiceFerc1TableTransformer(Ferc1AbstractTableTransformer):
+    """A transformer for the Electric Plant in Service table."""
+
+    table_id: Ferc1TableId = Ferc1TableId.PLANT_IN_SERVICE_FERC1
+
+    def get_row_map(self) -> pd.DataFrame:
+        """Construct a row map for a DBF table based on stored metadata."""
+        # Read the minimal, manually compiled row map for this table:
+        row_map = pd.read_csv(
+            importlib.resources.open_text("pudl.package_data.ferc1", "dbf_to_xbrl.csv"),
+            usecols=["sched_table_name", "report_year", "row_number", "xbrl_column"],
+        )
+        row_map = row_map.loc[
+            row_map.sched_table_name == self.source_table_id(Ferc1Source.DBF),
+            ["report_year", "row_number", "xbrl_column"],
+        ]
+
+        # Create an index containing all possible combinations of years and row numbers
+        idx = pd.MultiIndex.from_product(
+            [Ferc1Settings().dbf_years, row_map.row_number.unique()],
+            names=["report_year", "row_number"],
+        )
+
+        # Concatenate the row map with the empty index, so we have blank spaces to fill:
+        row_map = pd.concat(
+            [pd.DataFrame(index=idx), row_map.set_index(["report_year", "row_number"])],
+            axis="columns",
+        ).reset_index()
+
+        # Forward fill missing XBRL column names, until a new definition for the row
+        # number is found:
+        row_map["xbrl_column"] = row_map.groupby("row_number").xbrl_column.transform(
+            "ffill"
+        )
+        # Drop any rows that do not actually map between DBF rows and XBRL columns:
+        row_map = row_map.replace({"HEADER_ROW": np.nan}).dropna(subset=["xbrl_column"])
+        return row_map
+
+    @cache_df(key="dbf")
+    def process_dbf(self, raw_dbf: pd.DataFrame) -> pd.DataFrame:
+        """Process DBF inputs, aligning historical row numbers to XBRL account IDs."""
+        logger.info(f"{self.table_id.value}: Processing DBF data pre-concatenation.")
+        return (
+            self.drop_footnote_columns_dbf(raw_dbf)
+            # Note: in this rename_columns we have to pass in params, since we're using
+            # the inherited method, with param specific to the child class.
+            .pipe(self.rename_columns, params=self.params.rename_columns_ferc1.dbf)
+            .pipe(self.assign_record_id, source_ferc1=Ferc1Source.DBF)
+            .pipe(self.align_row_numbers_dbf)
+            .pipe(self.drop_unused_original_columns_dbf)
+            .pipe(
+                self.assign_utility_id_ferc1,
+                source_ferc1=Ferc1Source.DBF,
+            )
+        )
+
+    @cache_df(key="dbf")
+    def align_row_numbers_dbf(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Align historical FERC1 DBF row numbers with XBRL account IDs."""
+        return pd.merge(
+            df,
+            get_ferc1_dbf_xbrl_glue(self.source_table_id(Ferc1Source.DBF)),
```

Comment:
To parametrize this method along the lines of the other transformers, we could define a pydantic model to wrap the `dbf_xbrl_glue`, and have it instantiated by reading the data out of the CSV upon instantiation of the parameter object

Reply:
I think this makes a lot of sense and is worth doing. but also it doesn't feel like a make or break upgrade.

Reply:
I think it'll be easy once we've gone through and parameterized things, and it'll give us an obvious place to validate the hand mapping and hopefully give a useful error message if there's something that's obviously incorrect. But not trying to deal with the parameterizations yet.